### PR TITLE
Add vertx-jwt module from Beefy-ts

### PR DESCRIPTION
**Pull request by pjgg**
Vertx-jwt migration

Migrate 302-quarkus-vertx-jwt from [Beefy-ts](https://github.com/quarkus-qe/beefy-scenarios/tree/main/302-quarkus-vertx-jwt) to quarkus-test-suite

FIX GH-94

### CODE REVIEW

File: security/vertx-jwt/pom.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            <artifactId>quarkus-redis-client</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-openapi</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
+            <scope>test</scope>
```
Comment (Sgitario) [inline security/vertx-jwt/pom.xml:33]: This dependency is not needed.

File: security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/AuthNConfig.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithName;
+
+@ConfigMapping(prefix = "authN")
+public interface AuthNConfig {
+    String alg();
+
+    String secret();
+
+    @WithName("tokenLiveSpanMin")
```
Comment (Sgitario) [inline security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/AuthNConfig.java:12]: As far I know, this is not working. This was fixed in beefy as part of this commit: https://github.com/quarkus-qe/beefy-scenarios/commit/cc3fe7348b9ae582890037ae27c83ce12ce9891b

File: security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import io.quarkus.jackson.ObjectMapperCustomizer;
+
+@Singleton
+public class VertxAuthObjectMapperCustomizer implements ObjectMapperCustomizer {
+
+    @Override
+    public void customize(ObjectMapper objectMapper) {
+        // TODO https://github.com/quarkusio/quarkus/issues/16555
```
Comment (Sgitario) [inline security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java:15]: The issue has been fixed, so this TODO should be addressed.

**Review by Sgitario — CHANGES_REQUESTED** [review]
Some requested changes.
Did you try to run the tests on OpenShift?

**Comment by Sgitario** [comment]
Pls, add the issue link to be autoclosed once merged.

File: security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java (hunk 3 above)
Comment (pjgg) [reply security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java]: The way that was resolved is fine but we have created a new issue in order to improve the usability

File: security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import io.quarkus.jackson.ObjectMapperCustomizer;
+
+@Singleton
+public class VertxAuthObjectMapperCustomizer implements ObjectMapperCustomizer {
+
+    @Override
+    public void customize(ObjectMapper objectMapper) {
+        // TODO https://github.com/quarkusio/quarkus/issues/18927
```
Comment (Sgitario) [inline security/vertx-jwt/src/main/java/io/quarkus/ts/security/vertx/config/VertxAuthObjectMapperCustomizer.java:15]: This issue has been fixed and we can now do this via properties. Pls, update the PR accordingly.

**Comment by Sgitario** [comment]
rerun tests

**Review by Sgitario — CHANGES_REQUESTED** [review]
lgtm! just a minor update.

**Review by Sgitario — APPROVED** [review]
lgtm!
Let's wait for the OpenShift run

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
Failures are unrelated to this PR:
``` 
12:20:22 [INFO] Quarkus QE TS: Config .............................. FAILURE [26:12 min]
12:20:22 [INFO] Quarkus QE TS: Security: Vertx + JWT ............... SUCCESS [06:48 min]
```

Merging

**Pull request merged**